File: solutions/csharp/cars-assemble/1/CarsAssemble.cs
static class AssemblyLine
{

    public static double SuccessRate(int speed) => speed switch
    {
        10 => 0.77,
        9 => 0.8,
        8 or 7 or 6 or 5 => 0.9,
        4 or 3 or 2 or 1 => 1,
        _ => 0
        
    };

    public static double ProductionRatePerHour(int speed) => (speed * 221) * SuccessRate(speed);

    public static int WorkingItemsPerMinute(int speed) => (int)(ProductionRatePerHour(speed) / 60);
}

File: solutions/java/cars-assemble/1/src/main/java/CarsAssemble.java
public class CarsAssemble {
    private int _carsPerHour = 221;
    
    public double productionRatePerHour(int speed) 
    {
        if (speed == 0) { return 0; }
        if (speed > 0 && speed <= 4) { return _carsPerHour * speed; }
        if (speed > 4 && speed <= 8) { return (_carsPerHour * speed) * 0.9; }
        if (speed == 9) { return (_carsPerHour * speed) * 0.8; }
        if (speed == 10) { return (_carsPerHour * speed) * 0.77; }
        return 0;
    }

    public int workingItemsPerMinute(int speed) {
        return (int)productionRatePerHour(speed) / 60;
    }
}
